From 3e8d3cdce43a665fb89b0f805777e603f3eb30a4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <karel.koci@nic.cz>
Date: Mon, 1 Apr 2019 07:42:18 +0200
Subject: [PATCH] btrfs-progs: add support for zstd
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This allows BTRFS encryption using zstd. Using zstd can increase
performance.

This change increases size of btrfs-progs only by 4K. The real effect on
size is only size of libzstd which is around 500K. This should be fine
on system requiring BTRFS so no variant without zstd was added.

Signed-off-by: Karel Kočí <karel.koci@nic.cz>
---
 utils/btrfs-progs/Makefile | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/utils/btrfs-progs/Makefile b/utils/btrfs-progs/Makefile
index e674a4abb..d7c008c94 100644
--- a/utils/btrfs-progs/Makefile
+++ b/utils/btrfs-progs/Makefile
@@ -7,7 +7,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=btrfs-progs
 PKG_VERSION:=4.20.2
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE:=$(PKG_NAME)-v$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=@KERNEL/linux/kernel/people/kdave/btrfs-progs
@@ -28,7 +28,7 @@ define Package/btrfs-progs
   SECTION:=utils
   CATEGORY:=Utilities
   SUBMENU:=Filesystem
-  DEPENDS:=+libattr +libuuid +zlib +libblkid +liblzo +libpthread
+  DEPENDS:=+libattr +libuuid +zlib +zstd +libblkid +liblzo +libpthread
   TITLE:=Btrfs filesystems utilities
   URL:=https://btrfs.wiki.kernel.org/
 endef
@@ -52,8 +52,7 @@ CONFIGURE_ARGS += \
 	--disable-backtrace \
 	--disable-convert \
 	--disable-documentation \
-	--disable-python \
-	--disable-zstd
+	--disable-python
 
 define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/include $(1)/usr/lib
-- 
2.21.0

